api/src/controllers/team.js

```diff
@@ -247,22 +244,22 @@ async function deployToGHPages(req, res, next) {
  * @returns 404: team is not found
  * @returns 400: team id is not in a valid hexadecimal format
  */
-function updateTeam(req, res, next) {
+// eslint-disable-next-line no-unused-vars
+async function updateTeam(req, res, next) {
   const { team_id: _id } = req.params;
   const team = req.body;
 
-  Team.findByIdAndUpdate(_id, team, {
+  const updatedTeam = await Team.findByIdAndUpdate(_id, team, {
     new: true,
     runValidators: true,
   })
-    .then((updatedTeam) => res.status(200).json(updatedTeam))
-    .catch((err) => next(fillErrorObject(500, 'Server error', [err])));
+  res.status(200).json(updatedTeam)
```

Comment:
Is there a possibility this will throw if the update was unsuccessful, in which case we should wrap with a `try-catch` and if an error occurs, fill the error object?

Reply:
I think it is what the error middleware is responsible for. If findByIdAndUpdate throws an error(which I think the error might be server error or anything else we could not handle, as the teamId is already been verified by the validateTeamId middleware before this function), next will be called with a default Error object provided by the Express router. In this case, a default error object (no the type of fillErrorObject we created) will be passed into the error middleware, and the error middleware will handle it.